btree/btree.c

```diff
@@ -6973,51 +6956,29 @@ static int64_t btree_close_tree_tick(struct m0_sm_op *smop)
 {
 	struct m0_btree_op *bop  = M0_AMB(bop, smop, bo_op);
 	struct td          *tree = bop->bo_arbor->t_desc;
-	struct nd          *node;
+	int                 check = 0;
 
 	switch (bop->bo_op.o_sm.sm_state) {
 	case P_INIT:
 		if (tree == NULL)
 			return M0_ERR(-EINVAL);
 
 		M0_PRE(tree->t_ref != 0);
+		/* Fallthrough to P_ACT*/
 
+	case P_ACT:
 		m0_rwlock_write_lock(&tree->t_lock);
 		if (tree->t_ref > 1) {
 			m0_rwlock_write_unlock(&tree->t_lock);
-			bnode_put(tree->t_root->n_op, tree->t_root);
-			tree_put(tree);
-			return P_DONE;
+			check = 1;
```

Comment:
The code should do the way it was before except for the fact that we do not time out any more and instead we keep waiting till the reference count goes to 0.

Reply:
Done